File: dotnet/Lexy.Poc/Lexy.Poc.Core/Language/FunctionParameters.cs
using System.Collections.Generic;
using Lexy.Poc.Core.Parser;

namespace Lexy.Poc.Core.Language
{
    public class FunctionParameters : ParsableNode
    {
        public IList<VariableDefinition> Variables { get; } = new List<VariableDefinition>();

        public FunctionParameters(SourceReference reference) : base(reference)
        {
        }

        public override IParsableNode Parse(IParserContext context)
        {
            var variableDefinition = VariableDefinition.Parse(context);
            if (variableDefinition != null)
            {
                Variables.Add(variableDefinition);
            }
            return this;
        }

        protected override IEnumerable<INode> GetChildren() => Variables;

        protected override void Validate(IValidationContext context)
        {
        }
    }
}

File: typescipt/src/language/functions/functionParameters.ts


export class FunctionParameters extends ParsableNode {
   public Array<VariableDefinition> Variables = list<VariableDefinition>(): new;

   public FunctionParameters(SourceReference reference) : base(reference) {
   }

   public override parse(context: IParseLineContext): IParsableNode {
     let variableDefinition = VariableDefinition.Parse(VariableSource.Parameters, context);
     if (variableDefinition != null) Variables.Add(variableDefinition);
     return this;
   }

   public override getChildren(): Array<INode> {
     return Variables;
   }

   protected override validate(context: IValidationContext): void {
   }
}
